qiskit/quantum_info/operators/operator.py

```diff
@@ -403,27 +405,38 @@ def from_circuit(
                 initial_layout=layout,
                 input_qubit_mapping={qubit: index for index, qubit in enumerate(circuit.qubits)},
             )
+
+        initial_layout = layout.initial_layout if layout is not None else None
+
         if final_layout is None:
             if not ignore_set_layout and layout is not None:
                 final_layout = getattr(layout, "final_layout", None)
 
-        qargs = None
-        # If there was a layout specified (either from the circuit
-        # or via user input) use that to set qargs to permute qubits
-        # based on that layout
-        if layout is not None:
-            physical_to_virtual = layout.initial_layout.get_physical_bits()
-            qargs = [
-                layout.input_qubit_mapping[physical_to_virtual[physical_bit]]
-                for physical_bit in range(len(physical_to_virtual))
-            ]
-        # Convert circuit to an instruction
-        instruction = circuit.to_instruction()
-        op._append_instruction(instruction, qargs=qargs)
-        # If final layout is set permute output indices based on layout
+        from qiskit.synthesis.permutation.permutation_utils import _inverse_pattern
```

Comment:
kind of nit: Since we are always going to execute this line of code, shouldn't this be at the top of the function? Right after the docstring?

Reply:
`qiskit.synthesis` is a logically higher part of the package hierarchy than `qiskit.quantum_info`, but this part of `Operator` is acting above its position, so we use a scoped import to avoid import cycles.

Reply:
Oh I see. But my concern is more towards moving this to top of the function `from_circuit` and not the top of the file (which is where I think it could run into the circular import error). I might be wrong, can you please clarify?